Source/scrollrt.cpp

```diff
@@ -382,8 +382,8 @@ void DrawMonster(const Surface &out, int x, int y, int mx, int my, const Monster
 	int frames = SDL_SwapLE32(frameTable[0]);
 	if (nCel < 1 || frames > 50 || nCel > frames) {
 		const char *szMode = "unknown action";
-		if (monster._mmode <= 17)
-			szMode = MonsterModeNames[monster._mmode];
+		if (static_cast<int>(monster._mmode) <= 17)
+			szMode = MonsterModeNames[static_cast<int>(monster._mmode)];

```

Comment:
Could we turn this in to a switch-return function instead of and array with an enum as key?

Reply:
Optional, but greatly appreciated.